plans/sep8.md

```diff
@@ -65,6 +57,19 @@ class ApprovalProvider {
   }) => Promise<ActionPromptStatus>;
 }
 
+type callActionUrl = (params: CallActionUrlRequest) => string;
+
+interface CallActionUrlRequest {
+  response: ActionRequired;
+  callbackId: ApprovalRequest;
```

Comment:
❓ What's the motivation for renaming `request` to `callbackId`?

Reply:
Still learning, but my guess is that this field will be used by the issuer as an identifier when the issuer completes the additional work on their end to redirect the user back the where she was. In other words, the issuer might concatenate this field with the callbackUrl. The actual request body to the action URL will be built based on the returned information in the `ActionRequired` response. The action URL itself is in the `ActionRequired` response too. So I renamed it to avoid confusion.

I might be completely wrong about the purpose of having the ApprovalRequest here in the first place.

Reply:
[SEP-8](https://stellar.org/protocol/sep-8) has no callbacks. 

I think it'd be a good idea to understand what it is for before renaming and refactoring.

If this SEP-8 implementation supports callbacks then it implements more than SEP-8 defines today, and that isn't really SEP-8. Is callbacks something you're designing as a proposal for SEP-8?

Reply:
> Is callbacks something you're designing as a proposal for SEP-8?

Nope. The callback was in the [plan](https://github.com/stellar/js-stellar-wallets/blob/e2a9e31491da2ec61d1502acef8b033c749e2ae7/plans/sep8.md) before I started to modify it. I use it as a learning opportunity of how wallet SDK should be designed. Not so much related to SEP-8 before I understand the whole logic. And that's why I asked for your help in case I need to further modify SEP-8. 😄 

I think the reason why the callback was there in the first place is because of the [redirection](https://www.oauth.com/oauth2-servers/redirect-uris/redirect-uris-native-apps/) in native app.
```
     /**
       * On e.g. react native, the client will have to open a webview manually
       * and pass a callback URL that the app has "claimed." This is very similar
       * to e.g. OAuth flows.
       * https://www.oauth.com/oauth2-servers/redirect-uris/redirect-uris-native-apps/
       * Include the original request so it can be provided as a querystring to
       * the callback URL.
       */
```